data.lua

```diff
@@ -400,8 +400,13 @@ require "prototypes/creatures/caravan"
 require "prototypes/creatures/flying-caravan"
 require "prototypes/creatures/nuka-caravan"
 data.raw.unit["caravan"].affected_by_tiles = false
+if settings.startup["py-enable-fluid-caravans"].value then
```

Comment:
Setting-conditional requires used to cause desyncs and remain bad practice. Better to hide and disable the results.